src/v/utils/tests/fragmented_vector_test.cc

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "random/generators.h"
+#include "utils/fragmented_vector.h"
+
+#include <boost/test/unit_test.hpp>
+
+#include <vector>
+
+template<typename T>
+static void
+test_equal(std::vector<T>& truth, fragmented_vector<T, 1024>& other) {
+    BOOST_REQUIRE(!truth.empty());
+    BOOST_REQUIRE_EQUAL(truth.empty(), other.empty());
+    BOOST_REQUIRE_EQUAL(truth.size(), other.size());
+    BOOST_REQUIRE_EQUAL(truth.back(), other.back());
+    for (size_t x = 0; x < truth.size(); x++) {
+        BOOST_REQUIRE_EQUAL(truth[x], other[x]);
+    }
+}
+
+BOOST_AUTO_TEST_CASE(fragmented_vector_test) {
```

Comment:
Commit message is a little bit misleading

Reply:
fragmented buffer -> fragmented vector. thanks